docs/adr/0003-swift-packaging.md

```diff
@@ -0,0 +1,225 @@
+    Rust code as a binary artiact.
+
+## Decision Outcome
+
+Chosen option: **(D) Distribute multiple source-based Swift Packages, with pre-compiled Rust code**.
+
+This option will provide the best long-term consumer experience for iOS developers, and
+has the potential to simplify maintenance for application-services developers after an
+initial investment of effort.
+
+### Positive Consequences
+
+* Swift packages are very convenient to consume in newer versions of XCode.
+* Different iOS apps can choose to import a different subset of the available components,
+  potentiallying helping keep application size down.
+* Avoids issues with mis-matched Swift version between application-services build
+  and consumers, since Swift files are distributed in source form.
+* Encourages better conceptual separation between Swift code for different components;
+  e.g. it will make it possible for two Swift components to define an item of the same
+  name without conflicts.
+* Reduces the need to use XCode as part of application-services build process, in favour
+  of command-line tools.
+
+### Negative Consequences
+
+* More up-front work to move to this new setup.
+* May be harder to concurrently publish a Carthage framework for current consumers who aren't
+  able to move to Swift packages.
+* There is likely to be some amount of API breakage for existing consumers, if only in having
+  to replace a single `import MozillaAppServices` with independent imports of each component.
+
+### Implementation Sketch
+
+We will maintain the existing Carthage build infrastructure in the application-services repo and continue publishing a pre-built Carthage framework, to support firefox-ios until they migrate to Swift Packages.
+
+We will add an additional iOS build task in the application-services repo, that builds *just the Rust code* as a `.xcframework` bundle. An initial prototype shows that this can be achieved using a relatively straightforward shell script, rather than requiring a second XCode project. It will be
+published as a `.zip` artifact on each release in the same way as the current Carthage framework.
+
+We will create a new repository `rust-components-swift` to serve as the root of the new Swift Package
+distribution. It will import the `application-services` repository as a git submodule. This will let
+us iterate quickly on the Swift packaging setup without impacting existing consumers.
+
+The `rust-components-swift` repo will contain a `Package.swift` file that defines:
+
+* A single binary target that references the pre-built `.xcframework` bundle of Rust code.
+* One Swift target for each component, that references the Swift code from the git submodule
+  and depends on the pre-built Rust code.
+
+We will add automation to the `rust-components-swift` repo so that it automatically tracks
+releases made in the `application-services` repo and creates a corresponding git tag for
+the Swift package.
```

Comment:
For now it's probably fine if we still follow the a-s releases. Letting Glean releases trigger new packages can be considered in the future.